scalyr_agent/copying_manager/worker.py

```diff
@@ -1215,7 +1215,83 @@ def create_shared_object_manager(worker_class, worker_proxy_class):
     """
 
     class _SharedObjectManager(multiprocessing.managers.SyncManager):
-        pass
+        # the PID of the parent process.
+        parent_pid = None
+        # When the shared object manager's process is going to be terminated,
+        # we need to be sure that the worker is stopped first.
+        # This class attribute should provide access to the worker for the watchdog that is implemented below.
+        worker = None
+
+        @classmethod
+        def _create_worker(cls, *args, **kwargs):
+            """
+            Create a new worker and save it as a 'cls.worker' class attribute
+            to be able to access the worker's instance by the watchdog.
+            :return: the proxy object for the 'cls.worker'
+            """
+            cls.worker = worker_class(*args, **kwargs)
```

Comment:
I don't think it's right to have this be a class variable.  But, I think all of this can be simplified by using a separate watchdog abstraction as discussed above.

Reply:
I've managed to make it an instance variable, but anyway, we still have to bind the `shared object manager's` instance with the worker in order to be able to stop the worker in case of troubles.